src/components/PaymentInvoiceFilter.js

```diff
@@ -0,0 +1,156 @@
+import React from "react";
+import { injectIntl } from "react-intl";
+import _debounce from "lodash/debounce";
+
+import { Grid } from "@material-ui/core";
+import { withTheme, withStyles } from "@material-ui/core/styles";
+
+import { withModulesManager, formatMessage, TextInput, NumberInput, PublishedComponent } from "@openimis/fe-core";
+import { CONTAINS_LOOKUP, DEFAULT_DEBOUNCE_TIME, STARTS_WITH_LOOKUP } from "../constants";
+import PaymentInvoiceStatusPicker from "../pickers/PaymentInvoiceStatusPicker";
+
+const styles = (theme) => ({
+  form: {
+    padding: 0,
+  },
+  item: {
+    padding: theme.spacing(1),
+  },
+});
+
+const PaymentInvoiceFilter = ({ intl, classes, filters, onChangeFilters }) => {
+  const debouncedOnChangeFilters = _debounce(onChangeFilters, DEFAULT_DEBOUNCE_TIME);
+
+  const filterValue = (filterName) => filters?.[filterName]?.value;
+
+  const filterTextFieldValue = (filterName) => (filters[filterName] ? filters[filterName].value : "");
+
+  const onChangeFilter = (filterName) => (value) => {
+    debouncedOnChangeFilters([
+      {
+        id: filterName,
+        value: !!value ? value : null,
+        filter: `${filterName}: ${value}`,
+      },
+    ]);
+  };
+
+  const onChangeStringFilter =
```

Comment:
I think we could rename this function to sth like onChangeStringFilterGenerator, as it's used to setup actual filtering function, not adding filter directly.

Reply:
Yes, good idea, but due to some another priorities we can improve this later (this onChangeStringFilter is quite common - it is used in another frontend modules, we can think about generic way of creating such utils for filtering)